iib/workers/tasks/build.py

```diff
@@ -117,6 +118,16 @@ def _build_image(dockerfile_dir: str, dockerfile_name: str, request_id: int, arc
         {'cwd': dockerfile_dir},
         exc_msg=f'Failed to build the container image on the arch {arch}',
     )
+    log.debug('Verifying that %s was built with expected arch %s', destination, arch)
+    buildah_inspect = json.loads(run_cmd(['buildah', 'inspect', destination]))
+    archmap = {"amd64": "x86_64", "arm64": "aarch64", "s390x": "s390x", "ppc64le": "ppc64le"}
+    destination_arch = buildah_inspect['Docker']['config']['Labels']['architecture']
+
+    if destination_arch not in archmap.values() or destination_arch != archmap.get(arch):
+        log.warning("Wrong arch created for %s", destination)
+        exp_message = f'Wrong arch created, for image {destination} expected arch {arch}, \
+            found {destination_arch}'
+        raise IIBError(exp_message)
```

Comment:
Since the wrong arch is indeed being built by buildah, I wonder if it makes sense to raise `ExternalServiceError ` instead of `IIBError`.

Reply:
hmm, I was hoping to use IIBerror for the exception message, I can try that with ExternalServiceError